tools/pythonpkg/duckdb/experimental/spark/sql/functions.py

```diff
@@ -80,6 +81,44 @@ def struct(*cols: Column) -> Column:
     )
 
 
+def array(
+    *cols: Union["ColumnOrName", Union[List["ColumnOrName"], Tuple["ColumnOrName", ...]]]
+) -> Column:
+    """Creates a new array column.
+
+    .. versionadded:: 1.4.0
+
+    .. versionchanged:: 3.4.0
+        Supports Spark Connect.
+
+    Parameters
+    ----------
+    cols : :class:`~pyspark.sql.Column` or str
+        column names or :class:`~pyspark.sql.Column`\\s that have
+        the same data type.
+
+    Returns
+    -------
+    :class:`~pyspark.sql.Column`
+        a column of array type.
+
+    Examples
+    --------
+    >>> df = spark.createDataFrame([("Alice", 2), ("Bob", 5)], ("name", "age"))
+    >>> df.select(array('age', 'age').alias("arr")).collect()
+    [Row(arr=[2, 2]), Row(arr=[5, 5])]
+    >>> df.select(array([df.age, df.age]).alias("arr")).collect()
+    [Row(arr=[2, 2]), Row(arr=[5, 5])]
+    >>> df.select(array('age', 'age').alias("col")).printSchema()
+    root
+     |-- col: array (nullable = false)
+     |    |-- element: long (containsNull = true)
+    """
+    if len(cols) == 1 and isinstance(cols[0], (list, set)):
+        cols = cols[0]
+    return _invoke_function_over_columns("list_value", *cols)
```

Comment:
in duckdb there is a difference between array and list, `array` being a fixed size list.
I'm not sure spark has the same distinction? Just thought I'd mention it, that this *might* be confusing depending on what system the user is expecting the behavior of

Reply:
Agree, it's confusing if someone would expect DuckDB behaviour. As the goal is to have compatibility with the PySpark API, I hope that's not the case but something to keep an eye out! In my understanding, Spark's `ArrayType` is what are `list`s in DuckDB and there is no equivalent fixed-size version of it.